src/contexts/ErrorContext/ErrorContext.tsx

```diff
@@ -1,85 +1,69 @@
 import { logger } from "helpers";
-import {
-  PropsWithChildren,
-  createContext,
-  useCallback,
-  useContext,
-} from "react";
+import { ReactNode } from "react";
 import { useModalContext } from "../ModalContext";
 
-type State = {
-  handleError: (
-    error: any,
-    displayMessage?: string,
-    options?: { log: boolean }
-  ) => void;
-};
+function parseError(error: unknown): string | undefined {
+  if (typeof error === "string") return error;
 
-const Context = createContext<State>({
-  handleError: (_: any, _1?: string, _2?: { log: boolean }) => {},
-});
+  if (typeof error === "object" && error != null) {
+    if ("message" in error) return parseError(error.message);
+    if ("data" in error) return parseError(error.data);
+    if ("error" in error) return parseError(error.error);
+  }
 
-export default function ErrorContext(props: PropsWithChildren<{}>) {
+  if (error instanceof Error) return error.message;
+}
+
+export function useErrorContext() {
   const { showModal } = useModalContext();
 
   /**
-   * biome-ignore lint/correctness/useExhaustiveDependencies: bug in Biome with recursive dependencies
-   * Created an issue on their GH, see https://github.com/biomejs/biome/issues/2361
+   * @description for expected errors
+   * @param message - user can do something about
+   * @param error - if provided, would be logged to sentry for further investigation
    */
-  const handleError: State["handleError"] = useCallback(
-    (error: any, displayMessage?: string, options = { log: true }) => {
-      if (options.log) {
-        logger.error(error);
-      }
+  function displayError(message: ReactNode, error?: unknown) {
+    showModal(Prompt, { type: "error", children: message });
+    if (error) {
+      logger.error(error);
+    }
+  }
 
-      if (displayMessage) {
-        showModal(Prompt, {
-          type: "error",
-          children: displayMessage,
-        });
-      } else if (typeof error === "string") {
-        showModal(Prompt, {
-          type: "error",
-          children: error,
-        });
-      } else if (instanceOfAPError(error.data)) {
-        handleError(error.data);
-      } else if ("message" in error) {
-        handleError(error.message);
-        //TODO: specify controlled error shapes
-      } else if (error.data && typeof error.data === "string") {
-        handleError(error.data);
-      } else if (error.data && "message" in error.data) {
-        handleError(error.data.message);
-      } else if ("error" in error) {
-        handleError(error.error);
-      } else {
-        showModal(Prompt, {
-          type: "error",
-          children: GENERIC_ERROR_MESSAGE,
-        });
-      }
-    },
-    [showModal]
-  );
+  type Generic = {
+    /**
+     * context in present participle
+     * @example sending message
+     * @example loading resouce
+     */
+    context?: string;
+  };
+  type DisplayType = "parsed" | Generic | { custom: ReactNode };
```

Comment:
```suggestion
  type DisplayType = "parse" | Generic | { custom: ReactNode };
```
"parseD" (passive voice) implies the message has already been parsed/processed;
"parse" (active voice) implies a desired action.

Reply:
in the context of display `Type` - the word `parsed` is used as an adjective. i.e. "I want to display the `parsed` error" instead of the `generic` - or I want the `custom` type.